dask_glm/algorithms.py

```diff
@@ -134,7 +135,22 @@ def newton(X, y, max_iter=50, tol=1e-8, family=Logistic):
     return beta
 
 
-def admm(X, y, lamduh=0.1, rho=1, over_relax=1,
+def add_reg_grad(func):
+    @functools.wraps(func)
+    def wrapped(beta, X, y, z, u, rho):
+        return func(beta, X, y) + rho * (beta - z + u)
+    return wrapped
+
+
+def add_reg_f(func):
+    @functools.wraps(func)
+    def wrapped(beta, X, y, z, u, rho):
+        return func(beta, X, y) + (rho / 2) * np.dot(beta - z + u,
+                                                     beta - z + u)
+    return wrapped
+
+
+def admm(X, y, reg=L1, lamduh=0.1, rho=1, over_relax=1,
```

Comment:
I was thinking of elevating that option to the top-level API, just so I didn't have the check `if reg in ['l1', 'l2]` floating around in multiple places.